cognite/client/_api/agents/agents.py

```diff
@@ -290,22 +297,60 @@ def chat(
                 ...         Message("Once you have found it, find related time series.")
                 ...     ]
                 ... )
+
+            Chat with client-side actions:
+
+                >>> from cognite.client.data_classes.agents import ClientToolAction, ClientToolResult
+                >>> add_numbers_action = ClientToolAction(
+                ...     name="add",
+                ...     description="Add two numbers together",
+                ...     parameters={
+                ...         "type": "object",
+                ...         "properties": {
+                ...             "a": {"type": "number", "description": "First number"},
+                ...             "b": {"type": "number", "description": "Second number"},
+                ...         },
+                ...         "required": ["a", "b"]
+                ...     }
+                ... )
+                >>> response = client.agents.chat(
+                ...     agent_id="my_agent",
+                ...     messages=Message("What is 42 plus 58?"),
+                ...     actions=[add_numbers_action]
+                ... )
+                >>> if response.action_calls:
+                ...     for call in response.action_calls:
+                ...         # Execute the action
+                ...         result = call.arguments["a"] + call.arguments["b"]
+                ...         # Send result back
+                ...         response = client.agents.chat(
+                ...             agent_id="my_agent",
+                ...             messages=ClientToolResult(
+                ...                 action_id=call.action_id,
+                ...                 content=f"The result is {result}"
+                ...             ),
+                ...             cursor=response.cursor,
+                ...             actions=[add_numbers_action]
+                ...         )
         """
         self._warnings.warn()
 
         # Convert single message to list
-        if isinstance(messages, Message):
+        if isinstance(messages, (Message, ActionResult)):
```

Comment:
Using an Agent? This is an old syntax. It should be
```suggestion
        if isinstance(messages, Message | ActionResult):
```